File: peek_plugin_branch/tuples/BranchDetailTuple.py
from datetime import datetime

from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_branch._private.PluginNames import branchTuplePrefix


@addTupleType
class BranchDetailTuple(Tuple):
    """ Branch Tuple

    This tuple is a create example of defining classes to work with our data.
    """
    __tupleType__ = branchTuplePrefix + 'BranchDetailTuple'

    LOCATION_SERVER_BRANCH = 1
    LOCATION_LOCAL_BRANCH = 2

    #: Model Set Key
    modelSetKey: str = TupleField()

    #: Key
    # 
    # The key of this branch
    key: str = TupleField()

    #: Location
    # 
    # The location of this branch
    location: int = TupleField()

    #: Name
    # 
    # The location of this branch
    name: str = TupleField()

    #: Comment
    # 
    # The location of this branch
    comment: str = TupleField()

    #: User
    # 
    # The location of this branch
    user: str = TupleField()

    #: Updated Date
    # 
    # The location of this branch
    updatedDate: datetime = TupleField()

    #: Created Date
    # 
    # The location of this branch
    createdDate: datetime = TupleField()

File: peek_plugin_branch/plugin-module/BranchDetailTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {branchTuplePrefix} from "./_private/PluginNames";


/** Diagram Branch Service Enum
 *
 * This enum describes the location of the branch,
 * is the branch local to the UI, or is it persisted on the server.
 *
 */
export enum BranchLocation {
    ServerBranch = 1,
    LocalBranch = 2
}

@addTupleType
export class BranchDetailTuple extends Tuple {
    public static readonly tupleName = branchTuplePrefix + "BranchDetailTuple";


    /** The database ID
     *
     * Consider this field private
     */
    id: number;


    /** Model Set Key */
    modelSetKey: string;

    /** Key
     *
     * The key of this branch
     */
    key: string;

    /** Location
     *
     * The location of this branch
     */
    location: BranchLocation;

    /** Name
     *
     * The location of this branch
     */
    name: string;

    /** Comment
     *
     * The location of this branch
     */
    comment: string;

    /** userName
     *
     * The location of this branch
     */
    userName: string;

    /** Updated Date
     *
     * The location of this branch
     */
    updatedDate: Date;

    /** Created Date
     *
     * The location of this branch
     */
    createdDate: Date;

    constructor() {
        super(BranchDetailTuple.tupleName)
    }
}